### PR TITLE
Change appearance and command type

### PR DESCRIPTION
**Pull request by Mahesh1772**

### CODE REVIEW

File: src/test/java/data/TaskManagerTest.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ void updateTodo_validInput_updatesTask() throws TaskManagerException {
         addTask(date, initialTaskDescription, testTaskType, dummyTestDates);
 
         // Act
-        updateTask(date, 0, updatedTaskDescription);
+        //updateTask(date, 0, updatedTaskDescription);
 
         // Assert
         assertEquals(updatedTaskDescription, taskManager.getTasksForDate(date).get(0).getName());
```
Comment (clarencepohh) [inline src/test/java/data/TaskManagerTest.java:62]: This test seems to be failing. Could you check what changes you made that is causing it to fail?

Comment (clarencepohh) [reply src/test/java/data/TaskManagerTest.java:62]: On closer look, line 59 is commented out which may be the reason why the task description is not being updated

File: src/main/java/data/TaskManager.java [hunk 2]
```diff
@@ -313,39 +325,31 @@ private static void listTasksAtDate(TaskManager taskManager, LocalDate date, Str
...
-     * @param scanner User input.
      * @param weekView Current week being viewed.
      * @param inMonthView Whether month is being viewed.
      * @param taskManager The taskManager class being used.
      * @throws TaskManagerException If not in correct week/month view
      * @throws DateTimeParseException If there is an error parsing the date.
      */
-    public static void deleteManager(Scanner scanner, WeekView weekView, boolean inMonthView, TaskManager taskManager)
-            throws DateTimeParseException, TaskManagerException {
+    public static void deleteManager(WeekView weekView, boolean inMonthView, TaskManager taskManager, int day, int taskIndex)
```
Comment (clarencepohh) [inline src/main/java/data/TaskManager.java:334]: Line exceeds checkstyle length

File: src/main/java/data/TaskManager.java [hunk 3]
```diff
@@ -214,41 +204,63 @@ public static TaskType parseTaskType(String userInput) {
...
+//            System.out.println("Task updated.");
+//        } catch (NumberFormatException e) {
+//            System.out.println("Task number should be an integer value. Please try again.");
+//        } catch (IndexOutOfBoundsException e) {
+//            System.out.println("The task number you have entered does not exist. Please try again.");
+//        }
+//
+//    }
+
+    public static void updateManager(Scanner scanner, WeekView weekView, boolean inMonthView, TaskManager taskManager,
```
Comment (clarencepohh) [inline src/main/java/data/TaskManager.java:242]: Comment indentation is causing checkstyle test to fail. Also large segments of commented out code is bad coding practice. Do rectify

File: src/main/java/data/TaskManager.java [hunk 4]
```diff
@@ -126,55 +129,55 @@ public List<Task> getTasksForDate(LocalDate date) {
...
+        // Parse the task type
+        TaskType taskType = parseTaskType(taskTypeString.toUpperCase());
 
-            addTask(date, taskDescription, taskType, startAndEndDates);
-        } else {
-            String[] dummyDates = {null}; // dummy String array to pass into function call
-            addTask(date, taskDescription, taskType, dummyDates);
+        if (taskType == null) {
+            throw new TaskManagerException("Invalid task type. Please provide valid task type: " +
+                    "T for Todo, E for event, D for deadline.");
```
Comment (NicholasTanYY) [inline src/main/java/data/TaskManager.java:169]: Nice handling of the case where the user did not enter the task type

File: src/main/java/ui/UiRenderer.java [hunk 5]
```diff
@@ -101,5 +109,26 @@ public static void printSeparator() {
...
+        System.out.println("| - 'add, <day>, <taskType>, <taskDescription>': Add a new task.                |");
+        System.out.println("| - 'delete, <day>, <taskIndex>': Delete a task.                                |");
+        System.out.println("| - 'month': Switch to month view.                                              |");
+        System.out.println("| - 'week': Switch to week view.                                                |");
+        System.out.println("| - 'quit': Exit the calendar application.                                      |");
+        System.out.println(emptyLine);
+        System.out.println(horizontalLine);
+    }
+
+
```
Comment (NicholasTanYY) [inline src/main/java/ui/UiRenderer.java:132]: Neat!

File: src/main/java/time/MonthView.java [hunk 6]
```diff
@@ -25,56 +25,71 @@ public MonthView(LocalDate startOfMonth, DateTimeFormatter dateFormatter) {
...
+            if (current.getMonth().equals(YearMonth.from(startOfView).getMonth())) {
+                printDay(current);
+            } else {
+                System.out.print("|            ");
             }
-            System.out.println("|");
+            current = current.plusDays(1);
+        }
+        System.out.println("|");
+        printSeparator();
```
Comment (NicholasTanYY) [inline src/main/java/time/MonthView.java:52]: Probably can separate this out into more methods to prevent arrowhead codes.

**Review by NicholasTanYY — COMMENTED** [review]
Overall, good job on the exception handling and implementation of the single line commands! LGTM.

File: src/main/java/data/TaskManager.java [hunk 7]
```diff
@@ -81,25 +80,28 @@ public void deleteTask(LocalDate date, int taskIndex) {
...
+            throws IndexOutOfBoundsException, TaskManagerException {
         try {
             List<Task> dayTasks = tasks.get(date);
-            boolean dayHasTasks = dayTasks != null;
-            boolean taskIndexExists = taskIndex >= 0 && taskIndex < Objects.requireNonNull(dayTasks).size();
-            assert dayTasks != null;
-            assert taskIndexExists;
+            if (dayTasks == null || taskIndex < 0 || taskIndex >= dayTasks.size()) {
+                throw new TaskManagerException("The task you are trying to update does not exist.");
+            }
```
Comment (Aak242) [inline src/main/java/data/TaskManager.java:104]: Good job on handling potential exceptions with the existing TaskManagerException class!

File: src/main/java/data/TaskManager.java [hunk 8]
```diff
@@ -126,55 +129,55 @@ public List<Task> getTasksForDate(LocalDate date) {
...
 
-        if (inMonthView) {
-            checkIfDateInCurrentMonth(date);
+        // Convert the day to a LocalDate
+        LocalDate date;
+        // Convert the dayString to date range
+        String[] dates = new String[2];
+        if (day.contains("-")) {
+            dates = day.split("-");
         } else {
```
Comment (Aak242) [inline src/main/java/data/TaskManager.java:147]: This shorthand form of using days and hypens to add tasks quickly is sure to be useful for more experienced users

File: src/main/java/seedu/duke/Main.java [hunk 9]
```diff
@@ -78,22 +72,49 @@ public static void main(String[] args) throws IOException, TaskManagerException
                 break;
             case "update":
                 try {
-                    updateManager(scanner, weekView, inMonthView, taskManager);
-                } catch (TaskManagerException | DateTimeParseException e) {
+                    String[] parts = input.split(",\\s*");
+                    if (parts.length < 4) {
+                        throw new TaskManagerException("Invalid input format. Please provide input in the format: " +
+                                "update, <day>, <taskIndex>, <newDescription>");
+                    }
```
Comment (Aak242) [inline src/main/java/seedu/duke/Main.java:79]: Generally well documented error handling for the user to be able to quickly correct their format

File: src/main/java/ui/UiRenderer.java [hunk 10]
```diff
@@ -24,9 +25,16 @@ public class UiRenderer {
 
     private static final int numberOfDaysInWeek = 7;
 
-    public static void printWeekHeader() {
+    private static final String ANSI_RESET = "\u001B[0m";
+    private static final String ANSI_CYAN = "\u001B[36m"; // Cyan
```
Comment (Aak242) [inline src/main/java/ui/UiRenderer.java:29]: Good use of final variables to make colour codes more readable

File: src/main/java/ui/UiRenderer.java [hunk 11]
```diff
@@ -101,5 +109,26 @@ public static void printSeparator() {
...
+        System.out.println("| - 'prev': Move to the previous week or month view.                            |");
+        System.out.println("| - 'update, <day>, <taskIndex>, <newDescription>': Update a task description.  |");
+        System.out.println("| - 'add, <day>, <taskType>, <taskDescription>': Add a new task.                |");
+        System.out.println("| - 'delete, <day>, <taskIndex>': Delete a task.                                |");
+        System.out.println("| - 'month': Switch to month view.                                              |");
+        System.out.println("| - 'week': Switch to week view.                                                |");
+        System.out.println("| - 'quit': Exit the calendar application.                                      |");
+        System.out.println(emptyLine);
+        System.out.println(horizontalLine);
+    }
```
Comment (Aak242) [inline src/main/java/ui/UiRenderer.java:130]: Detailed help page for users, along with the correct format to be used

**Review by Aak242 — COMMENTED** [review]
LGTM!

**Pull request merged**